lib/cassava/client.rb

```diff
@@ -13,15 +13,26 @@ def initialize(session, opts = {})
     # @see #insert
     def insert_async(table, data)
       ttl = data.delete(:ttl)
-      executor.execute_async(insert_statement(table, data, ttl), :arguments => data.values)
+      consistency = data.delete(:consistency)
+      statement = insert_statement(table, data, ttl)
+      if consistency.nil?
+        executor.execute_async(statement, :arguments => data.values)
+      else
+        executor.execute_async(statement, { :arguments => data.values, :consistency => consistency })
+      end
     end
 
     # @param table [Symbol] the table name
     # @param data [Hash] A hash of column names to data, which will be inserted into the table
     def insert(table, data)
       ttl = data.delete(:ttl)
+      consistency = data.delete(:consistency)
```

Comment:
Pattern following? What's the reason for this particular mechanism of bringing the data/options to code usefulness? Asking for a friend